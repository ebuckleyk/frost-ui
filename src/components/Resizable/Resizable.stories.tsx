import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';

import { ResizableHandle, ResizablePanel, ResizablePanelGroup } from './Resizable';

function ResizableDemo() {
  return (
    <ResizablePanelGroup direction="horizontal" className="max-w-md rounded-lg border">
      <ResizablePanel defaultSize={50}>
        <div className="flex h-[200px] items-center justify-center p-6">
          <span className="font-semibold">One</span>
        </div>
      </ResizablePanel>
      <ResizableHandle />
      <ResizablePanel defaultSize={50}>
        <ResizablePanelGroup direction="vertical">
          <ResizablePanel defaultSize={25}>
            <div className="flex h-full items-center justify-center p-6">
              <span className="font-semibold">Two</span>
            </div>
          </ResizablePanel>
          <ResizableHandle />
          <ResizablePanel defaultSize={75}>
            <div className="flex h-full items-center justify-center p-6">
              <span className="font-semibold">Three</span>
            </div>
          </ResizablePanel>
        </ResizablePanelGroup>
      </ResizablePanel>
    </ResizablePanelGroup>
  );
}
function VerticalResizeDemo() {
  return (
    <ResizablePanelGroup direction="vertical" className="min-h-[200px] max-w-md rounded-lg border">
      <ResizablePanel defaultSize={25}>
        <div className="flex h-full items-center justify-center p-6">
          <span className="font-semibold">Header</span>
        </div>
      </ResizablePanel>
      <ResizableHandle />
      <ResizablePanel defaultSize={75}>
        <div className="flex h-full items-center justify-center p-6">
          <span className="font-semibold">Content</span>
        </div>
      </ResizablePanel>
    </ResizablePanelGroup>
  );
}

function WithHandleDemo() {
  return (
    <ResizablePanelGroup direction="horizontal" className="min-h-[200px] max-w-md rounded-lg border">
      <ResizablePanel defaultSize={25}>
        <div className="flex h-full items-center justify-center p-6">
          <span className="font-semibold">Sidebar</span>
        </div>
      </ResizablePanel>
      <ResizableHandle withHandle />
      <ResizablePanel defaultSize={75}>
        <div className="flex h-full items-center justify-center p-6">
          <span className="font-semibold">Content</span>
        </div>
      </ResizablePanel>
    </ResizablePanelGroup>
  );
}

type ComponentType = React.ComponentProps<typeof ResizablePanelGroup>;
const meta: Meta<ComponentType> = {
  component: ResizablePanelGroup,
  subcomponents: { ResizableHandle, ResizablePanel },
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {
  render: ResizableDemo,
};

export const Vertical: Story = {
  render: VerticalResizeDemo,
};

export const WithHandle: Story = {
  render: WithHandleDemo,
};
