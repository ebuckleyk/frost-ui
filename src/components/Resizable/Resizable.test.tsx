import * as React from 'react';
import { render } from '@testing-library/react';

import { ResizableHandle, ResizablePanel, ResizablePanelGroup } from './Resizable';

function Component() {
  return (
    <ResizablePanelGroup direction="horizontal" className="max-w-md rounded-lg border">
      <ResizablePanel defaultSize={50}>
        <div className="flex h-[200px] items-center justify-center p-6">
          <span className="font-semibold">One</span>
        </div>
      </ResizablePanel>
      <ResizableHandle />
      <ResizablePanel defaultSize={50}>
        <ResizablePanelGroup direction="vertical">
          <ResizablePanel defaultSize={25}>
            <div className="flex h-full items-center justify-center p-6">
              <span className="font-semibold">Two</span>
            </div>
          </ResizablePanel>
          <ResizableHandle />
          <ResizablePanel defaultSize={75}>
            <div className="flex h-full items-center justify-center p-6">
              <span className="font-semibold">Three</span>
            </div>
          </ResizablePanel>
        </ResizablePanelGroup>
      </ResizablePanel>
    </ResizablePanelGroup>
  );
}

describe('Resizable', () => {
  it('should render Resizable component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
