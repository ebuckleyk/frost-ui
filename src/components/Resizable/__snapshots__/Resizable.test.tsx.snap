// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resizable > should render Resizable component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full w-full data-[panel-group-direction=vertical]:flex-col max-w-md rounded-lg border"
        data-panel-group=""
        data-panel-group-direction="horizontal"
        data-panel-group-id=":r0:"
        style="display: flex; flex-direction: row; height: 100%; overflow: hidden; width: 100%;"
      >
        <div
          class=""
          data-panel=""
          data-panel-group-id=":r0:"
          data-panel-id=":r1:"
          data-panel-size="50.0"
          style="flex-basis: 0px; flex-grow: 50; flex-shrink: 1; overflow: hidden;"
        >
          <div
            class="flex h-[200px] items-center justify-center p-6"
          >
            <span
              class="font-semibold"
            >
              One
            </span>
          </div>
        </div>
        <div
          class="relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90"
          data-panel-group-direction="horizontal"
          data-panel-group-id=":r0:"
          data-panel-resize-handle-enabled="true"
          data-panel-resize-handle-id=":r2:"
          data-resize-handle=""
          data-resize-handle-state="inactive"
          role="separator"
          style="user-select: none;"
          tabindex="0"
        />
        <div
          class=""
          data-panel=""
          data-panel-group-id=":r0:"
          data-panel-id=":r3:"
          data-panel-size="50.0"
          style="flex-basis: 0px; flex-grow: 50; flex-shrink: 1; overflow: hidden;"
        >
          <div
            class="flex h-full w-full data-[panel-group-direction=vertical]:flex-col"
            data-panel-group=""
            data-panel-group-direction="vertical"
            data-panel-group-id=":r4:"
            style="display: flex; flex-direction: column; height: 100%; overflow: hidden; width: 100%;"
          >
            <div
              class=""
              data-panel=""
              data-panel-group-id=":r4:"
              data-panel-id=":r5:"
              data-panel-size="25.0"
              style="flex-basis: 0px; flex-grow: 25; flex-shrink: 1; overflow: hidden;"
            >
              <div
                class="flex h-full items-center justify-center p-6"
              >
                <span
                  class="font-semibold"
                >
                  Two
                </span>
              </div>
            </div>
            <div
              class="relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90"
              data-panel-group-direction="vertical"
              data-panel-group-id=":r4:"
              data-panel-resize-handle-enabled="true"
              data-panel-resize-handle-id=":r6:"
              data-resize-handle=""
              data-resize-handle-state="inactive"
              role="separator"
              style="user-select: none;"
              tabindex="0"
            />
            <div
              class=""
              data-panel=""
              data-panel-group-id=":r4:"
              data-panel-id=":r7:"
              data-panel-size="75.0"
              style="flex-basis: 0px; flex-grow: 75; flex-shrink: 1; overflow: hidden;"
            >
              <div
                class="flex h-full items-center justify-center p-6"
              >
                <span
                  class="font-semibold"
                >
                  Three
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full w-full data-[panel-group-direction=vertical]:flex-col max-w-md rounded-lg border"
      data-panel-group=""
      data-panel-group-direction="horizontal"
      data-panel-group-id=":r0:"
      style="display: flex; flex-direction: row; height: 100%; overflow: hidden; width: 100%;"
    >
      <div
        class=""
        data-panel=""
        data-panel-group-id=":r0:"
        data-panel-id=":r1:"
        data-panel-size="50.0"
        style="flex-basis: 0px; flex-grow: 50; flex-shrink: 1; overflow: hidden;"
      >
        <div
          class="flex h-[200px] items-center justify-center p-6"
        >
          <span
            class="font-semibold"
          >
            One
          </span>
        </div>
      </div>
      <div
        class="relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90"
        data-panel-group-direction="horizontal"
        data-panel-group-id=":r0:"
        data-panel-resize-handle-enabled="true"
        data-panel-resize-handle-id=":r2:"
        data-resize-handle=""
        data-resize-handle-state="inactive"
        role="separator"
        style="user-select: none;"
        tabindex="0"
      />
      <div
        class=""
        data-panel=""
        data-panel-group-id=":r0:"
        data-panel-id=":r3:"
        data-panel-size="50.0"
        style="flex-basis: 0px; flex-grow: 50; flex-shrink: 1; overflow: hidden;"
      >
        <div
          class="flex h-full w-full data-[panel-group-direction=vertical]:flex-col"
          data-panel-group=""
          data-panel-group-direction="vertical"
          data-panel-group-id=":r4:"
          style="display: flex; flex-direction: column; height: 100%; overflow: hidden; width: 100%;"
        >
          <div
            class=""
            data-panel=""
            data-panel-group-id=":r4:"
            data-panel-id=":r5:"
            data-panel-size="25.0"
            style="flex-basis: 0px; flex-grow: 25; flex-shrink: 1; overflow: hidden;"
          >
            <div
              class="flex h-full items-center justify-center p-6"
            >
              <span
                class="font-semibold"
              >
                Two
              </span>
            </div>
          </div>
          <div
            class="relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90"
            data-panel-group-direction="vertical"
            data-panel-group-id=":r4:"
            data-panel-resize-handle-enabled="true"
            data-panel-resize-handle-id=":r6:"
            data-resize-handle=""
            data-resize-handle-state="inactive"
            role="separator"
            style="user-select: none;"
            tabindex="0"
          />
          <div
            class=""
            data-panel=""
            data-panel-group-id=":r4:"
            data-panel-id=":r7:"
            data-panel-size="75.0"
            style="flex-basis: 0px; flex-grow: 75; flex-shrink: 1; overflow: hidden;"
          >
            <div
              class="flex h-full items-center justify-center p-6"
            >
              <span
                class="font-semibold"
              >
                Three
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
